mlrun/runtimes/local.py

```diff
@@ -442,7 +446,20 @@ def exec_from_params(handler, runobj: RunObject, context: MLClientCtx, cwd=None)
         try:
             if cwd:
                 os.chdir(cwd)
-            val = handler(**kwargs)
+            # Apply the MLRun handler decorator for parsing inputs using type hints and logging outputs using log hints
+            # (Only if needed - returns or type hints were given by the user):
+            val = (
+                mlrun.handler(
+                    inputs=(
+                        runobj.spec.inputs_type_hints
+                        if runobj.spec.inputs_type_hints
+                        else True
+                    ),
+                    outputs=runobj.spec.returns if runobj.spec.returns else False,
+                )(handler)(**kwargs)
+                if runobj.spec.inputs_type_hints or runobj.spec.returns
+                else handler(**kwargs)
+            )
```

Comment:
```suggestion
            if runobj.spec.inputs_type_hints or runobj.spec.returns:
                val = handler(**kwargs)
             else:
              val = (
                  mlrun.handler(
                      inputs=(
                          runobj.spec.inputs_type_hints
                          if runobj.spec.inputs_type_hints
                          else True
                      ),
                      outputs=runobj.spec.returns if runobj.spec.returns else False,
                  )(handler)(**kwargs)
```
I would write something like this ^^, I think its clearer 

Also now noticing its quite weird that inputs / outputs receives both dictionaries and bool please explain

Reply:
changed to your preference. 

regarding the question, see the docs of the `mlrun.handler`:

**inputs**: Parsing configurations for the arguments passed as inputs via the `run` method of an MLRun function. Can be passed as a boolean value or a dictionary:

* `True` - Parse all found inputs to the assigned type hint in the function's signature. If there is no type hint assigned, the value will remain an `mlrun.DataItem`.
* `False` - Do not parse inputs, leaving the inputs as `mlrun.DataItem`.
* `Dict[str, Union[Type, str]]` - A dictionary with argument name as key and the expected type to parse the `mlrun.DataItem` to. The expected type can be a string as well, idicating the full module path.

Default: `True`.

**outputs**: Logging configurations for the function's returned values. Expecting a list of values:

* `str` - A string in the format of '{key}:{artifact_type}'. If a string was given without ':' it will indicate the key and the artifact type will be according to the returned value type. The artifact types can be one of: "dataset", "directory", "file", "object", "plot" and "result".
* `Dict[str, str]` - A dictionary of logging configuration. the key 'key' is mandatory for the logged artifact key.
 * `None` - Do not log the output.

The list length must be equal to the total amount of returned values from the function. Default is `None` - meaning no outputs will be logged.

Reply:
regarding the question, see the docs of the `mlrun.handler`:

**inputs**: Parsing configurations for the arguments passed as inputs via the `run` method of an MLRun function. Can be passed as a boolean value or a dictionary:

* `True` - Parse all found inputs to the assigned type hint in the function's signature. If there is no type hint assigned, the value will remain an `mlrun.DataItem`.
* `False` - Do not parse inputs, leaving the inputs as `mlrun.DataItem`.
* `Dict[str, Union[Type, str]]` - A dictionary with argument name as key and the expected type to parse the `mlrun.DataItem` to. The expected type can be a string as well, idicating the full module path.

Default: `True`.

**outputs**: Logging configurations for the function's returned values. Expecting a list of values:

* `str` - A string in the format of '{key}:{artifact_type}'. If a string was given without ':' it will indicate the key and the artifact type will be according to the returned value type. The artifact types can be one of: "dataset", "directory", "file", "object", "plot" and "result".
* `Dict[str, str]` - A dictionary of logging configuration. the key 'key' is mandatory for the logged artifact key.
 * `None` - Do not log the output.

The list length must be equal to the total amount of returned values from the function. Default is `None` - meaning no outputs will be logged.